spec/jobs/hyrax/autopopulation/approval_job_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Hyrax::Autopopulation::ApprovalJob, type: :job do
+  describe "create_work" do
+    context "using hyrax" do
+      let(:work) { create(:work, title: ["work"], autopopulation_status: "draft") }
+      let(:work_1) { create(:work, title: ["work_1"], autopopulation_status: "draft") }
+      let(:work_2) { create(:work, title: ["work_2"], autopopulation_status: "draft") }
+
+      let(:email) { "admin@example.com" }
+      let(:user) { build_stubbed(:user, email: email) }
+      let(:approve_multiple_works) { ["multiple", user, [work.id]] }
+      let(:approve_all_works) { ["all", user] }
+
+      before do
+        allow(User).to receive(:find).with(user.id.to_s).and_return(user)
+      end
+
+      it "submitted ids can be used to approve works" do
+        described_class.perform_now(*approve_multiple_works)
+        expect(work.reload.autopopulation_status).to eq("approved")
+      end
+
+      it "all works with draft status can be approved" do
+        expect { described_class.perform_now(*approve_all_works) }.to change { work_1.reload.autopopulation_status }.from("draft").to("approved")
```

Comment:
What happens when we have works with different statuses? Can we have a test that checks that only some of them are updated?

Reply:
We have that test already:
https://github.com/ubiquitypress/hyrax-autopopulation/blob/f39101fc8909150bfa49de899257129451e67638/spec/jobs/hyrax/autopopulation/approval_job_spec.rb#L21-L24

Reply:
As I read it, that test moves 3 works in draft all to approved.  What if one work was already approved or had another status? (Are there any other statuses?)

Reply:
If it is already approved, it will remain approved. There are only two possible status, that is draft or approved. 
I will add a test to cover that.